File: Python/2024/5. May/Codeforces/F. 3SUM.py
# DATE ==> 12 May 2024
# AUTHOR ==> SUMONTO 

for _ in range(int(input())):
    n = int(input())
    arr = list(map(int, input().split(" ")))
    a = [i%10 for i in arr]
    d = list(set(a))
    for i in range(len(set(a))):
        if a.count(d[i]) >= 3:
            d.append(d[i])
            d.append(d[i])
        if len(d) == 30: break

    n = len(d)
    ans = False
    for a in range(n):
        for b in range(a+1, n):
            for c in range(b+1, n):
                if (d[a]+d[b]+d[c])%10 == 3 and a!=b and a!=c and b!=c:
                    ans = True
                    break
            if ans: break
        if ans: break
    print("YES" if ans else "NO")

File: C , C++/2024/5. May/Codeforces/f_3sum.cpp
// DATE ==> 1 May 2024
// Problem Name ==> F. 3SUM Codeforces Round 799 (Div. 4)
// Problem Rating ==> 1300
// AUTHOR ==> SUMONTO
#include <bits/stdc++.h>
using namespace std;

int main() {
    int TEST;
    cin >> TEST;
    while(TEST--) {
        int N;
        cin >> N;
        vector<int> vec;
        for(int i = 0; i < N; ++i) {
            int x;
            cin >> x;
            vec.push_back(x%10);
        }

        bool found = false;
        for(int i = 0; i < N; ++i) {
            for(int j = i + 1; j < N; ++j) {
                for(int k = j + 1; k < N; ++k) {
                    int sum = vec[i] + vec[j] + vec[k];
                    if (sum % 10 == 3) {
                        found = true;
                        break; 
                    }
                }
                if (found) break; 
            }
            if (found) break; 
        }

        if (found) {
            cout << "YES" << endl;
        } else {
            cout << "NO" << endl;
        }
    }
    return 0;

    // TIME ERROR TIME ERROR TIME ERROR TIME ERROR TIME ERROR TIME ERROR //
    // TIME ERROR TIME ERROR TIME ERROR TIME ERROR TIME ERROR TIME ERROR //
    // TIME ERROR TIME ERROR TIME ERROR TIME ERROR TIME ERROR TIME ERROR //
    // TIME ERROR TIME ERROR TIME ERROR TIME ERROR TIME ERROR TIME ERROR //
    // TIME ERROR TIME ERROR TIME ERROR TIME ERROR TIME ERROR TIME ERROR //
    // TIME ERROR TIME ERROR TIME ERROR TIME ERROR TIME ERROR TIME ERROR //
    // TIME ERROR TIME ERROR TIME ERROR TIME ERROR TIME ERROR TIME ERROR //
    // TIME ERROR TIME ERROR TIME ERROR TIME ERROR TIME ERROR TIME ERROR //


}